python_modules/dagster/dagster/_core/execution/backfill.py

```diff
@@ -109,7 +113,24 @@ def selector_id(self):
 
     @property
     def is_asset_backfill(self) -> bool:
-        return self.serialized_asset_backfill_data is not None
+        return (
+            self.serialized_asset_backfill_data is not None or self.asset_backfill_data is not None
+        )
+
+    def get_asset_backfill_data(self, asset_graph: AssetGraph) -> Optional[AssetBackfillData]:
```

Comment:
I find the behavior of this function a little unexpected with respect to its signature. Just based on the signature, I would expect it to return `None` in the case where there is no backfill data, not in the case where there's a deserialization issue. Thoughts on still handling the error inside the caller instead of here?

Seems like we then might be able to use this at [1] instead of having to directly call `from_serialized` there?